### DISCUSSION TITLE
A freed object should be treated as a "false" value

### DISCUSSION THREAD

**Discussion by qt911025** [godotengine/godot-proposals#9843] [Scripting]
A freed object is a <Freed Object>, also treated as a null value.
For some types, there is no way to convert it into a bool, but it can be asserted directly, which turn out to be an opposite result.
```gdscript
extends Node

@onready var _sub_node := $Node/SubNode

func _ready() -> void:
	$Node.queue_free()
	await get_tree().process_frame

	prints("_sub_node is null", _sub_node == null)
	if _sub_node:
		prints("_sub_node is true")
	else:
		prints("_sub_node is false")
```
So if I want to check existence of an object, I should use `if _sub_node != null` rather than `if _sub_node`. It's counterintuitive, isn't it?

**Comment by Zireael07** [comment]
It used to be directly comparable to false but was changed to be more explicit.

Also: use if_instance_valid(_sub_node)

**Comment by AThousandShips** [comment]
This was done by design, see:
* https://github.com/godotengine/godot/issues/59816

And as suggested above the proper way is to use `is_instance_valid`, see the [documentation](https://docs.godotengine.org/en/stable/classes/class_object.html#description)

**Reply by AThousandShips** [reply]
To udpate, this *is* the case unless there's been some error in the code, my bad got it confused with `freed == null` which is *not* true

**Comment by dalexeev** [comment]
* See godotengine/godot#90513.
